docs/adrs/0073-database-exports.md

```diff
@@ -0,0 +1,206 @@
+for handling the datasets. There is a second option which streams datasets
+directly to S3. This option is _the streaming option_. The non-streaming option
+at the time of this writing will require some minor changes to our configuration
+for ECS tasks. This change has the potential to affect the MilMove SSAG in order
+to support ECS Fargate engine tasks from running both as a non-root user and
+with a non-writable root file system.
+
+### Low complexity and is easily updated in the future
+
+We don't want to build solutions which would require an extensive rewrite of the
+solution proposed for this ADR. This means that any alternative here will need
+to be weighed against the level of effort it would cause the teams to restart
+the work from scratch. This means that we will need to consider alternatives
+which help achieve the goals of incremental changes using a custom CDC solution.
+**This does not mean that the chosen alternative needs to implement these
+changes. This is mostly to be able to talk about the pros and cons of a given
+solution.**
+
+### Doesn't negatively impact the security of the application
+
+The alternatives here must not impact the security of the application
+negatively. Currently our ECS tasks run within a non-root and unprivileged user.
+This is further secured by our disabling of having a writable root file system.
+In order to bypass these restrictions, some of the alternatives introduce the
+need of mounted Bind Points or AWS EFS volumes. These technical details will
+require more infrastructure and additional changes to our SSAG document
+depending on the alternative chosen.
+
+### In conclusion
+
+In conclusion, the alternatives presented below are weighed by their pros 🟩
+and cons 🟥. Out of the decision outcomes section, we will choose a proposed
+solution and place it under the **Proposal:** heading. Within this section, we
+will go into further technical details and possible implementation details.
+These extra details are not done for all of the proposed alternatives. Instead
+we rely on the pros and cons of the alternatives to relay that information.
+
+## Considered Alternatives
+
+**Bold text is the chosen alternative**.
+
+- Using Lambda or ECS: Save the dataset to a file system then upload to S3
+- Using Lambda or ECS: Stream the dataset directly to S3 in-memory
+- Have AWS save an RDS Snapshot automatically and place in S3 bucket
+- **Have AWS DMS save datasets directly into S3 as a data migration target**
+
+## Decision Outcome
+
+The most promising alternative in this ADR is the AWS DMS alternative. It uses
+CDC transactions for extracting the datasets from the database, can be run on an
+interval using <TECHNOLOGY_NAME>, and transforms the CDC datasets into CSV files
+and automatically places them in a structured layout within a target S3 bucket.
+
+> Flesh out the above to answer the question about: How do we define intervals
+> for the AWS DMS changes?
```

Comment:
from talking to @CoryAtTruss, here's some additional context.

> You can set up a DMS task for either one-time migration or on-going replication. An on-going replication task keeps your source and target databases in sync. Once set up, the on-going replication task will continuously apply source changes to the target with minimal latency. All DMS features, such as data validation and transformations, are available for any replication task.

Reply:
If we happen to need an automated DMS migration task, here's a blog post by AWS that discusses how to do this with Cron and ECS/EC2.

=> https://aws.amazon.com/blogs/database/automating-aws-dms-migration-tasks/